feat: adding recipes tracker workshop (#56456)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/workshop-recipe-tracker/66fbcf750a62784cf11f5634.md

---
id: 66fbcf750a62784cf11f5634
title: Step 10
challengeType: 1
dashedName: step-10
---

# --description--

Time to test each of the functions. You can use any of the recipes for this, but go with `recipe1` for now.

> Create a `recipe1AverageRating`, `recipe1TotalIngredients`, and `recipe1DifficultyLevel` variables . For their values, call the function that corresponds to each variable name and pass in the appropriate `recipe1` property.

Reviewer: ```suggestion
Create three new variables called `recipe1AverageRating`, `recipe1TotalIngredients`, and `recipe1DifficultyLevel`. For their values, call the function that corresponds to each variable name and pass in the appropriate `recipe1` property.
```

> Create three new variables called `recipe1AverageRating`, `recipe1TotalIngredients`, and `recipe1DifficultyLevel`. For their values, call the function that corresponds to each variable name and pass in the appropriate `recipe1` property.

Reviewer 2: ```suggestion
Create three new variables: `recipe1AverageRating`, `recipe1TotalIngredients`, and `recipe1DifficultyLevel`. Assign them the values by calling the corresponding function for each variable and passing in the appropriate `recipe1` property.
```

> Also, log each of the variables to the console to see what they look like.

Reviewer 2: ```suggestion
Finally, log each variable to the console to see the results.
```

# --hints--

You should create a `recipe1AverageRating` variable.

```js
assert.isNotNull(recipe1AverageRating)
```

Your `recipe1AverageRating` variable should be set to `getAverageRating(recipe1.ratings)`.

```js
assert.equal(recipe1AverageRating, getAverageRating(recipe1.ratings));
```

You should log `recipe1AverageRating` to the console with the text `"Recipe 1 Average Rating"` followed by the variable name.

```js
assert.match(code, /console\.log\((?:(`|"|')Recipe\s+1\s+Average\s+Rating\s*:\s*(`|"|')\s*\+\s*recipe1AverageRating|`Recipe\s+1\s+Average\s+Rating\s*:\s*\$\{recipe1AverageRating\}`)\);?/)

> ```

Reviewer: The directions don't mention logging the string followed by the variable name. But also, this didn't pass 

```md
console.log("Recipe 1 Average Rating",recipe1AverageRating) 
```

to make testing easier, we should just have them log the variables to the console which is what the directions say to do

You should create a `recipe1TotalIngredients` variable.

```js
assert.isNotNull(recipe1TotalIngredients)
```

Your `recipe1TotalIngredients` variable should be set to `getTotalIngredients(recipe1.ingredients)`.

```js
assert.equal(recipe1TotalIngredients, getTotalIngredients(recipe1.ingredients));
```

You should log `recipe1TotalIngredients` to the console with the text `"Recipe 1 Total Ingredients"` followed by the variable name.

```js
assert.match(code, /console\.log\((?:(`|"|')Recipe\s+1\s+Total\s+Ingredients:\s*(`|"|')\s*\+\s*recipe1TotalIngredients|`Recipe\s+1\s+Total\s+Ingredients:\s+\$\{recipe1TotalIngredients\}`)\);?/)

> ```

Reviewer: we should update this too based on the suggestions from earlier

You should create a `recipe1DifficultyLevel` variable.

```js
assert.isNotNull(recipe1DifficultyLevel)
```

Your `recipe1DifficultyLevel` variable should be set to `getDifficultyLevel(recipe1.cookingTime)`.

```js
assert.equal(recipe1DifficultyLevel, getDifficultyLevel(recipe1.cookingTime));
```

You should log `recipe1DifficultyLevel` to the console with the text `"Recipe 1 Total Ingredients"` followed by the variable name.

```js
assert.match(code, /console\.log\((?:(`|"|')Recipe\s+1\s+Difficulty\s+Level:\s*(`|"|')\s*\+\s*recipe1DifficultyLevel|`Recipe\s+1\s+Difficulty\s+Level:\s+\$\{recipe1DifficultyLevel\}`)\);?/)

> ```

Reviewer: we should update this too based on the suggestions from earlier
